jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/MessageInputStream.java

```diff
@@ -221,6 +247,11 @@ public int read() throws IOException
                         case CLOSED:
                             return -1;
 
+                        case COMPLETE:
+                            // If we are complete we have read the last frame but
+                            // don't want to resume reading until onMessage() exits.
```

Comment:
What?  I don't understand this comment.

So the application is using `InputStream.read(byte[], int, int)`, and the state is COMPLETE, meaning there's no more frames to read, why are we returning "written" and not -1?

Reply:
`COMPLETE` means it has all the content up to `EOF`.
`CLOSED` means all the content has been read up to `EOF` or `close()` has been called. (we will return -1 in `CLOSED` state).

I will add better comments to explain the difference.